tests/kafkatest/services/kafka/templates/kafka.properties

```diff
@@ -29,8 +29,6 @@ socket.request.max.bytes=104857600
 num.partitions=1
 num.recovery.threads.per.data.dir=1
 log.retention.hours=168
-log.segment.bytes=1073741824
-log.cleaner.enable=false
```

Comment:
Why do we have to do this?

Reply:
"kafka.properties" template file contains default config values to populate server.properties.

Now log cleaner is enabled by default. So I have removed this property to enable it by default is tests.  

"log.segment.bytes"  property configured with  sever default value and not updated by any of the test. 

 For now I reverted this change.  if required we can do this cleanup later.

Also updated the logic in kafka.py to cleanup the duplicates between kafka.properties and server_prop_overides. Updated the logic to load the kafka.properties first and then override any server_prop_overides configs.

Reply:
I think it's good to enable the cleaner in the system tests. And if the other property is just the default, I agree we should just remove it from here.